### PR TITLE
Components: Cleanup types

### PR DESCRIPTION
**Pull request by lifeiscontent**
Issue:

## What I did
updated types for components to be compatible with React 18, with one caveat, https://github.com/probablyup/markdown-to-jsx/issues/435

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: lib/components/package.json [hunk 1]
```diff
@@ -53,8 +54,12 @@
   },
   "devDependencies": {
     "@popperjs/core": "^2.6.0",
+    "@testing-library/jest-dom": "^5.16.4",
     "@types/color-convert": "^2.0.0",
+    "@types/lodash": "^4.14.182",
     "@types/overlayscrollbars": "^1.12.0",
+    "@types/qs": "^6.9.7",
+    "@types/util-deprecate": "^1.0.0",
```
Comment (lifeiscontent) [inline lib/components/package.json:62]: @shilman not sure if I should add these, but without them `tsc` was complaining.

File: lib/components/src/blocks/ArgsTable/TabbedArgsTable.tsx [hunk 2]
```diff
@@ -19,11 +20,9 @@ export const TabbedArgsTable: FC<TabbedArgsTableProps> = ({ tabs, ...props }) =>
...
         const [label, table] = entry;
         const id = `prop_table_div_${label}`;
-        return (
+        return ({ active }: { active: boolean }) => (
           <div key={id} id={id} title={label}>
-            {({ active }: { active: boolean }) =>
-              active ? <ArgsTable key={`prop_table_${label}`} {...table} {...props} /> : null
-            }
+            {active ? <ArgsTable key={`prop_table_${label}`} {...table} {...props} /> : null}
           </div>
```
Comment (lifeiscontent) [inline lib/components/src/blocks/ArgsTable/TabbedArgsTable.tsx:26]: not sure how to test this, but the types were wrong so I figured this is what the intent was.

File: lib/components/package.json [hunk 3]
```diff
@@ -37,7 +37,8 @@
     "*.d.ts"
   ],
   "scripts": {
-    "prepare": "ts-node ../../scripts/prebundle.ts"
+    "prepare": "ts-node ../../scripts/prebundle.ts",
+    "check": "tsc --noEmit"
```
Comment (lifeiscontent) [inline lib/components/package.json:41]: not sure if you guys have something similar, but this is a command I add to every package to make sure types are correct.

File: lib/components/package.json [hunk 4]
```diff
@@ -51,9 +52,13 @@
   },
   "devDependencies": {
     "@popperjs/core": "^2.6.0",
+    "@testing-library/jest-dom": "^5.16.4",
     "@types/color-convert": "^2.0.0",
+    "@types/lodash": "^4.14.182",
     "@types/overlayscrollbars": "^1.12.0",
     "@types/react-syntax-highlighter": "11.0.5",
+    "@types/qs": "^6.9.7",
+    "@types/util-deprecate": "^1.0.0",
```
Comment (lifeiscontent) [inline lib/components/package.json:61]: not sure if I should remove these or not, but `tsc` was complaining without them.

File: lib/components/package.json (hunk 3 above)
Comment (ndelangen) [reply lib/components/package.json:41]: yeah, I want to add that to all packages indeed!

File: lib/components/src/blocks/ArgsTable/TabbedArgsTable.tsx (hunk 2 above)
Comment (ndelangen) [reply lib/components/src/blocks/ArgsTable/TabbedArgsTable.tsx:26]: I think the types are incorrect here, chromatic picked up changes that don't look right.

Comment (ndelangen) [reply lib/components/src/blocks/ArgsTable/TabbedArgsTable.tsx:26]: @lifeiscontent could you revert these runtime changes, and find another method for making the types fit? 👏

Comment (lifeiscontent) [reply lib/components/src/blocks/ArgsTable/TabbedArgsTable.tsx:26]: @ndelangen do you know if the `div` element is always a `div`? or can it be any element? after that, I'll see if I can create a type that works for it, but I've never seen anything like that before as its telling a child element about its props which is pretty odd.

Comment (ndelangen) [reply lib/components/src/blocks/ArgsTable/TabbedArgsTable.tsx:26]: I'm pretty sure the type of element `div` or otherwise doesn't matter, no.

**Comment by lifeiscontent** [comment]
@ndelangen ready for review again

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
CI didn't run for some reason. I pushed in the base branch to trigger it.

**Comment by lifeiscontent** [comment]
@ndelangen are these failures flakes or actual issues?

**Comment by ndelangen** [comment]
The 2 failures are known issues, and aren't due to your changes, this is good to merge!

**Pull request merged**